_posts/2019-08-26-how-to-use-gruntwork-infrastructure-as-code-library.adoc

```diff
@@ -1399,6 +1409,272 @@ module "vpc" {
   <<automated_tests_terraform>>.
 . Deploy your changes to each environment. You do this using the same process outlined in <<deploy_terraform>>.
 
+[[integration_with_tfc_tfe]]
+=== Integrating with Terraform Cloud and Enterprise
+This section will cover how to use Gruntwork in conjunction with two popular HashiCorp products: Terraform Cloud (TFC)
+and Terraform Enterprise (TFE). Although the open source edition of Terraform is quite powerful and flexible as a
+standalone project, many organizations turn to TFC/TFE for the CLI/UI integration, approval-based workflow capabilities,
+Sentinel policy framework, and more.
+
+It's easy to use the production-grade, battle-tested Terraform modules from Gruntwork's infrastructure-as-code library
+to deploy infrastructure with TFC and TFE. With a little more effort, you can also use Terragrunt! Throughout this
+section, to avoid repeating ourselves, we'll only mention TFC. The same patterns work just as well with TFE.
+
+[NOTE]
+.Obtain an API token
+====
+Before working with TFC/TFE from your local system, you'll need to obtain an API token. You can generate an API token in your user settings of the TFC UI. Follow https://www.terraform.io/docs/commands/cli-config.html#credentials-1[the official instructions] to use the token.
+====
+
+==== Using a Gruntwork module in a TFC workspace
```

Comment:
I read through this section and to be honest, had a bit of trouble following it, even though me and you worked through this live just the other day!

I think (a) we need a bit more detail and (b) we should probably have more screenshots to guide the user here. There risk, of course, is that the TFC UI totally changes, and the screenshots are out of date, but I'd prefer a good experience now that needs updating later, rather than a crappy experience now and later.

Here's a rough outline of how I'd approach it:

**One-time setup**

1. Go to `app.terraform.io` (or your custom TFE URL).
1. Configure a VCS connection. Show screenshot creating an app in GitHub, then copying app details into TFC.
1. Add your SSH key. Show screenshot of associating SSH key with GitHub user and another copying the key into TFC.

**Deploy an example for learning**

(Cross-reference the examples, modules, test breakdown of our repos in this same doc)

1. Go to `app.terraform.io` (or your custom TFE URL).
1. Create workspace. (screenshot)
1. Pick VCS you added in initial setup and select a repo. We'll use package-sns (screenshot).
1. Enter a sub-folder. We want `examples/xxx`. (screenshot)
1. Enter variables. E.g. The name for the SNS topic (screenshot).
1. Enter env vars. E.g., AWS Access Keys and AWS region (screenshot).
1. Run `plan` (screenshot).
1. Run `apply` (screenshot). 
1. You have an SNS topic deployed! (screenshot)

**Deploy code for production**

1. Go to `app.terraform.io` (or your custom TFE URL).
1. Create a module in your `infra-modules` repo. E.g., Here's an example of a production SNS module.
1. Commit the changes: `git commit`, `git push`, etc.
1. Create workspace. (screenshot)
1. Pick VCS you added in initial setup and select a repo. This time, pick your `infra-modules` repo (screenshot).
1. Enter a sub-folder. We want `<path to your module in infra-modules>`. (screenshot)
1. Enter variables. E.g. The name for the SNS topic (screenshot).
1. Enter env vars. E.g., AWS Access Keys and AWS region (screenshot).
1. Run `plan` (screenshot).
1. Run `apply` (screenshot). 
1. You have an SNS topic deployed! (screenshot)

Reply:
I was trying to find a balance between an informative section on the topic and full fledged tutorial on using TFC. Sounds like you think the tutorial approach is going to be more helpful. I don't necessarily disagree, but I do wonder whether this guide is really the right place for such a tutorial. Should we make this a standalone guide? It would be relatively short in comparison to the other guides, but then wouldn't constitute a major section within this guide.